File: shipmentNotifier-spec/src/main/java/io/billie/shipmentnotifierspec/request/ShipmentRequest.java
package io.billie.shipmentnotifierspec.request;


import java.io.Serializable;
import java.math.BigDecimal;

import javax.validation.constraints.Min;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;

import lombok.Data;


@Data
public class ShipmentRequest implements Serializable {


	@NotBlank(message = "invalid.order.trackingCode")
	private String orderTrackingCode;

	@NotNull(message = "invalid.shipment.amount")
	@Min(value = 1, message = "shipment.amount.must.greater.zero")
	private BigDecimal shipmentAmount;

	@NotBlank(message = "invalid.merchant.number")
	private String merchantNumber;

}

File: src/main/kotlin/io/billie/organisations/orders/view/ShipmentRequest.kt
package io.billie.organisations.orders.view

import com.fasterxml.jackson.annotation.JsonProperty
import io.billie.organisations.orders.domain.Shipment
import java.math.BigDecimal

/**
 * Request object mapped to [Shipment] domain object
 */
data class ShipmentRequest(@JsonProperty("amount") val amount: BigDecimal)